packages/wrangler/src/cloudchamber/cli/index.ts

```diff
@@ -10,19 +10,55 @@ import { brandColor, dim } from "@cloudflare/cli/colors";
 import { spinner } from "@cloudflare/cli/interactive";
 import {
 	DeploymentsService,
+	ImageRegistriesService,
 	PlacementsService,
 	SshPublicKeysService,
 } from "../client";
 import { wrap } from "../helpers/wrap";
 import { idToLocationName } from "../locations";
 import type {
+	CustomerImageRegistry,
 	DeploymentV2,
 	ListSSHPublicKeys,
 	PlacementEvent,
 	PlacementWithEvents,
 } from "../client";
 import type { EventName, Status } from "../enums";
 
+export function pollRegistriesUntilCondition(
+	onRegistries: (registries: Array<CustomerImageRegistry>) => boolean
+): Promise<Array<CustomerImageRegistry>> {
+	return new Promise<Array<CustomerImageRegistry>>((res, rej) => {
+		let errCount = 0;
+		const poll = () => {
+			ImageRegistriesService.listImageRegistries()
+				.then((registries) => {
+					try {
+						if (!onRegistries(registries)) {
+							setTimeout(() => poll(), 500);
+						} else {
+							res(registries);
+						}
+					} catch (err) {
+						rej(err);
+					}
```

Comment:
So if the `onRegistries()` function throws we don't retry but reject immediately?

Reply:
@IRCody or @gabivlj any opinions on this?